### PR TITLE
Update TestController.psm1 SetGlobalVariables to add root user check

### PR DESCRIPTION
**Pull request by lixuemin2016**
## Description
Update TestController.psm1's function SetGlobalVariables to add root user check. If set as "root", exit the test framework directly. Not sure here is the best place to check vmUsername. XMLSecretFile also changes the vmUsername, it is called in PrepareTestEnviroment function of child class.

Issue #
If do not add root user check, it will get lots of error when running LisAv2 related with /home/xxx paths.
Although the main instruction guide demonstrates that do not use root user account. When new user uses LisAv2, they may ignore this requirement, then need long time to find the failure reason.

Thank you so much.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] Verify PR checker results and fix any coding errors.
- [ ] Included LISAv2 sample test results to demonstrate code functionality.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/LIS/LISAv2/blob/master/.github/CONTRIBUTING.md).

==========================================================
PS C:\Users\xuli\workspace\repo\LISAv2> .\Run-LisaV2.ps1 -TestPlatform HyperV -RGIdentifier 20190225 -OsVHD C:\RHEL-8.0.
0-20190213.0-x86_64-GEN1-A.vhdx -TestNames KVP-BASIC-CHECKS -VMGeneration 1
02/28/2019 09:51:59 : [INFO ] Autogenerated test ID: PY52
02/28/2019 09:52:00 : [INFO ] Found branch information and updated with  add_root_user_check
02/28/2019 09:52:00 : [INFO ] Path length is too long for current working directory 'C:\Users\xuli\workspace\repo\LISAv2
'
02/28/2019 09:52:00 : [INFO ] Copying current workspace to C:\LISAv2\PY52
02/28/2019 09:52:04 : [INFO ] Working directory has been changed to C:\LISAv2\PY52
02/28/2019 09:52:04 : [INFO ] Logging directory: C:\LISAv2\PY52\TestResults\2019-28-02-17-52-04-8558
02/28/2019 09:52:04 : [INFO ] Setting parameter: TestPlatform = HyperV
02/28/2019 09:52:04 : [INFO ] Setting parameter: RGIdentifier = 20190225
02/28/2019 09:52:04 : [INFO ] Setting parameter: OsVHD = C:\RHEL-A.vhdx
02/28/2019 09:52:04 : [INFO ] Setting parameter: TestNames = KVP-BASIC-CHECKS
02/28/2019 09:52:04 : [INFO ] Setting parameter: TestIterations = 1
02/28/2019 09:52:04 : [INFO ] Setting parameter: VMGeneration = 1
02/28/2019 09:52:04 : [INFO ] Test parameters for HyperV have been validated successfully. Continue running the test.
02/28/2019 09:52:04 : [ERROR] Failed to update configuration files. '-XMLSecretFile [FilePath]' is not provided.
02/28/2019 09:52:04 : [INFO ] Set Test Location to GlobalConfiguration.Global.HyperV.Hosts.ChildNodes[0].ServerName
02/28/2019 09:52:05 : [INFO ] Updated C:\LISAv2\PY52\XML\GlobalConfigurations.xml file.
02/28/2019 09:52:05 : [INFO ] ------------------------------------------------------------------
02/28/2019 09:52:05 : [INFO ] HyperV Host            : 2019-auto
02/28/2019 09:52:05 : [INFO ] Destination VHD Path   : VHDs_Destination_Path
02/28/2019 09:52:05 : [INFO ] ------------------------------------------------------------------
02/28/2019 09:52:05 : [INFO ] Setting global variables
02/28/2019 09:52:05 : [ERROR] EXCEPTION: Root user cannot be used in the test configuration
02/28/2019 09:52:05 : [ERROR] Source: Line 181 in script C:\Users\xuli\workspace\repo\LISAv2\TestControllers\TestControl
ler.psm1.
02/28/2019 09:52:05 : [INFO ] Copying all files back to original working directory: C:\Users\xuli\workspace\repo\LISAv2.
02/28/2019 09:52:17 : [INFO ] Setting workspace back to original location: C:\Users\xuli\workspace\repo\LISAv2
02/28/2019 09:52:17 : [INFO ] LISAv2 exit code: 1

### CODE REVIEW

**Comment by chvalean** [comment]
This is test-cases related, though in some platforms/scenarios I'm thinking you can have them run directly with the root account.
So we should not limit this, at best maybe throw a warning but not fail.

Anyhow, this should be covered as part of the README, and not controlled from automation end.
@juhlee-microsoft

**Comment by iamshital** [comment]
@chvalean Agreed. We have an upcoming requirement to use root user directly. 
@juhlee-microsoft @chvalean We can talk about this offline.

**Comment by juhlee-microsoft** [comment]
@lixuemin2016 I am not sure if I understand your PR. Accessing root user priv can be done in Test Case script level, and I am not sure why you put this error handling in TestController. If you need root user priv in test run-time, please refer other script like using runAsSudo parameter. If you are ok with my feedback, please close the PR.

**Comment by lixuemin2016** [comment]
@chvalean @juhlee-microsoft  @iamshital 
Thank you so much for all your review comments.
Let me explain more for this PR.

When tested on HyperV platform, I did not notice that "non-root" user should be set up, then configured LinxuUsername as "root" in the GlobalConfigurations.xml,  test case failed by lots of errors, since there was no /home/root folder. 

[Error] Failed to execute: dmesg > /home/root/initialBootLogs.txt
[Error] Unable to identify /home/root/state.txt
[Error]  dmesg >/home/root/FinalBootLogs.txt, 
[Error] /home/root/detectLinuxDistro.sh  
[Error] Doing Diffie-Hellman group exchange xxxxxxx
[WARN] Download error, attempt, Retrying for download

In other words,  it cannot use "root" account in the GlobalConfigurations.xml. Here "root" does not mean root user priv in the test case. if set up "root" account, I think that how about exiting the framework directly, or giving warning to avoid lots of error message?

Now I agree to close this PR, looks that have an upcoming requirement to use root user directly.
